hooks/ruby/dismiss-review-server/server.rb

```diff
@@ -0,0 +1,115 @@
+require 'sinatra'
+require 'json'
+require 'uri'
+require 'net/http'
+
+$github_api_token = ENV['GITHUB_API_TOKEN']
+$github_secret_token = ENV['SECRET_TOKEN']
+
+post '/payload' do
+
+  # Only validate secret token if set 
+  if !$github_secret_token.nil?
+    payload_body = request.body.read
+    verify_signature(payload_body)
+  end
+
+  github_event = request.env['HTTP_X_GITHUB_EVENT']
+  if github_event == "push"
+    request.body.rewind
+    parsed = JSON.parse(request.body.read)
+
+    # Get branch information
+    branch_head = parsed['ref']
+    branch_name = branch_head.chomp("refs/heads")
+    
+    # Get Repository owner
+    repo_owner = parsed["repository"]["owner"]["name"]
+
+    # Create URL to look up Pull Requests for this branch
+    # e.g. https://api.github.com/repos/baxterthehacker/public-repo/pulls{/number}
+    pulls_url = parsed['repository']['pulls_url']
+    
+    # Pull off the {/number}" and search for all Pull Requests
+    # that include the branch
+    pulls_url_filtered = pulls_url.split('{').first + "?head=#{repo_owner}:#{branch_name}"
```

Comment:
The PR is created on the upstream repository. This script won't work across forks because the webhook is likely running on the forked repo. With this being a 'sample', I'm fine with this limitation for now